cpp/src/parquet/stream_reader.cc

```diff
@@ -0,0 +1,258 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "parquet/stream_reader.h"
+
+namespace parquet {
+
+StreamReader::StreamReader(std::unique_ptr<ParquetFileReader>& File_Reader)
```

Comment:
this should either be a raw pointer or a non-reference (i.e. do a ownership transfer).  If a raw pointer it should be clarified in the documentation that ownership is not taken.  It might be nice to provide an option to do both for users.

Reply:
It would be dangerous to have a raw pointer.   Ownership is taken by having a unique_ptr parameter which is released.    Ownership makes sense given that the destructor of this class closes the file.
If ownership should not be taken, a shared_ptr could be passed.   In this case, the destructor would be changed not to close the file.